lib/openai/http_headers.rb

```diff
@@ -0,0 +1,34 @@
+module OpenAI
+  module HTTPHeaders
+    def headers
+      if azure?
+        azure_headers
+      else
+        openai_headers
+      end.merge(extra_headers)
+    end
+
+    def openai_headers
+      {
+        "Content-Type" => "application/json",
+        "Authorization" => "Bearer #{@access_token}",
+        "OpenAI-Organization" => @organization_id
+      }
+    end
+
+    def azure_headers
+      {
+        "Content-Type" => "application/json",
+        "api-key" => @access_token
+      }
+    end
+
+    def extra_headers
+      @extra_headers ||= {}
+    end
+
+    def add_headers(headers)
+      @extra_headers = extra_headers.merge(headers.transform_keys(&:to_s))
```

Comment:
Did you intend to override the extra headers here?  Seems non-obvious that this would happen if I set them explicitly on initialization.

Reply:
Yes - that seemed the easiest way to add headers without changing the existing API too much. I think it's reasonable to expect that if you do something like this the latest value should take precedence. Is this what you're worried about or is there something else I'm missing here?

```ruby
client = OpenAI::Client.new(extra_headers: { 'My-Header': 'a value' })
client.add_headers('My-Header': 'another value'))
```

Reply:
Yep exactly that. `extra_headers.merge(@extra_headers || {}).merge(headers.transform_keys(&:to_s))` or similar should do it I think?